traitsui/examples/demo/Standard_Editors/VideoEditor_demo.py

```diff
@@ -0,0 +1,158 @@
+
+
+def QImage_from_np(image):
+    assert (np.max(image) <= 255)
+    image8 = image.astype(np.uint8, order='C', casting='unsafe')
+    height, width, colors = image8.shape
+    bytesPerLine = 4 * width
+
+    image = QImage(image8.data, width, height, bytesPerLine,
+                   QImage.Format_RGB32)
+    return image
+
+
+def np_from_QImage(qimage):
+    # Creates a numpy array from a pyqt(5) QImage object
+    width, height = qimage.width(), qimage.height()
+    channels = qimage.pixelFormat().channelCount()
+    return np.array(
+        qimage.bits().asarray(width * height * channels)
+    ).reshape(height, width, channels).astype('u1')
+
+
+def qimage_function(antialiasing=True):
+    def antialis_fun(image_fun):
+        """
+        Turns a image funciton into a QImage function,
+        bound within the viewing frames box.
+        """
+        def qimage_conv_fun(image, box_dims):
+            _np_image = image_fun(np_from_QImage(image))
+            pil_image = Image.fromarray(_np_image)
+            if antialiasing:
+                pil_image.thumbnail(box_dims, Image.ANTIALIAS)
+            else:
+                pil_image.thumbnail(box_dims)
+            _np_image = np.array(pil_image)
+            image = QImage_from_np(_np_image)
+            return image, _np_image
+        return qimage_conv_fun
+    return antialis_fun
+
+
+@qimage_function(antialiasing=False)
+def test_image_fun(image):
+    return image.transpose(1, 0, 2)
+
+
+class VideoEditorDemo(HasTraits):
+    """Defines the main VideoEditor demo class."""
+
+    #: The URL that holds the video data.
+    video_url = Str()
+
+    #: A button that
+    play_pause_button = Button()
+
+    button_label = Str('Play')
+
+    state = PlayerState
+
+    duration = Float
+
+    position = Range(low=0.0, high='duration')
+
+    error = Str
+
+    status = MediaStatus
+
+    buffer = Range(0, 100)
+
+    muted = Bool(True)
+
+    volume = Range(0.0, 100.0, value=100.0)
+
+    playback_rate = Float(1.0)
+
+    image_fun = Callable()
+
+    def _state_changed(self, new):
+        if new == 'stopped' or new == 'paused':
+            self.button_label = 'Play'
+        elif new == 'playing':
+            self.button_label = 'Pause'
+
+    def _play_pause_button_changed(self):
+        if self.state == 'stopped' or self.state == 'paused':
+            self.state = 'playing'
+        else:
+            self.state = 'paused'
+
+    # Demo view
+    traits_view = View(
+        UItem('video_url'),
+        UItem(
+            'video_url',
+            editor=VideoEditor(
+                state='state',
+                position='position',
+                duration='duration',
+                video_error='error',
+                media_status='status',
+                buffer='buffer',
+                muted='muted',
+                volume='volume',
+                playback_rate='playback_rate',
+                image_fun='image_fun'
```

Comment:
It would be nice not to require users to provide traits for all of this editor state. How do the trait declarations in `VideoEditor` need to change?

Reply:
This editor was designed to be low-level and makes heavy use of the `sync_name` metadata, behaviour is explained here:
https://github.com/enthought/traitsui/blob/7f3dbaf6c30ed221e1aa680e0c9bb2a34c552b39/traitsui/editor.py#L594-L617

In particular users of the code are responsible for setting up any play/pause buttons, muting etc.  If they are left empty, then the default values from the Editor will be used, I think, so we should make sure that those are sensible.

If we want to allow constant values (eg. setting `muted` to `True` and not synchronising), then we might need to have `muted` and `muted_name` traits.

Reply:
> If they are left empty, then the default values from the Editor will be used

This is true, but since the default values are empty strings it causes a `TraitError` if nothing is passed.

I don't understand the mechanisms in TraitsUI well enough to change this [even after some experimentation], so I'm going to leave it alone in this PR.